### PR TITLE
Install ESLint, cleanup some code, add avatarUtil module

### PR DESCRIPTION
**Pull request by blms**
(Note: This branch is called `site-viz` but this PR doesn't concern site-level visualizations. It's work done in preparation for that.)

### Changes
- Install and configure ESLint [linting utility](https://eslint.org/docs/about/)
- Pull out duplicated class functions `this.getAvatar` into a reusable module called `avatarUtil`, per #22 
  - This is used in the following views: `Account.react.js`, `Admin.react.js`,`Config.react.js`, `Graph.react.js`, `Main.react.js`, `Study.react.js`
  - Fix linter errors/warnings in (or "lint", for short) all affected views—mostly cleanup of unused variables and imports
- Other code cleanup tasks like removing commented-out code

### Testing
- [x] Rebuilt all code (`npm run build`)
- [x] Rebuilt Singularity image (`dpdash-ben-0604.sif` in `/data/predict`)
- [x] Tested all functions in all affected views in the UI

### CODE REVIEW

**Comment by tashrifbillah** [comment]
> It's work done in preparation for that.

How?

File: views/Main.react.js [hunk 1]
```diff
@@ -501,16 +479,17 @@ class MainPage extends Component {
       body: JSON.stringify({
         preferences: preference
       })
-    }).then((response) => {
+    }).then(() => {
```
Comment (tashrifbillah) [inline views/Main.react.js:482]: Even though `response` variable is not used downstream, was it necessary to remove that?

File: views/Main.react.js [hunk 2]
```diff
@@ -501,16 +479,17 @@ class MainPage extends Component {
...
       return;
     });
   }
-  handleResize = (event) => {
```
Comment (tashrifbillah) [inline views/Main.react.js:508]: Same comment as of for `response`

File: views/Main.react.js [hunk 3]
```diff
@@ -534,24 +513,6 @@ class MainPage extends Component {
       return index % 2 === 0 ? 'evenRow' : 'oddRow'
     }
   }
-  getAvatar = () => {
```
Comment (tashrifbillah) [inline views/Main.react.js:537]: Again we are finding ourselves in a position where we have to update similar block of code in multiple places! I see `getAvatar = ()` function in many places.

**Comment by blms** [comment]
> > It's work done in preparation for that.
> 
> How?

I will use both ESLint and the avatar utility in producing the site-level visualizations.

File: views/Main.react.js (hunk 3 above)
Comment (tashrifbillah) [reply views/Main.react.js:537]: Ignore it, sorry!

Comment (blms) [reply views/Main.react.js:537]: That's why I removed the duplicated code from each view and pulled it out into its own module called `avatarUtil`.

File: views/Main.react.js (hunk 1 above)
Comment (blms) [reply views/Main.react.js:482]: Not strictly necessary but it was flagged by ESLint, and I wanted to eliminate all ESLint warnings possible. It is a cleanup task.

**Comment by tashrifbillah** [comment]
My visualizations don't use that. Did you brainstorm on how you can start from mine or how does this change will affect my visualizations? The work plan is not to do the same work twice, and definitely not something on I have spent a few months already.

**Comment by blms** [comment]
> My visualizations don't use that. Did you brainstorm on how you can start from mine or how does this change will affect my visualizations? The work plan is not to do the same work twice, and definitely not something on I have spent a few months already.

A few things:
- ESLint is not a dependency, rather it is a tool used to clean up code and catch errors before compile time.
- The avatar utility is used in displaying the DPDash sidebar, which I understood to be part of the task of integrating the visualizations.
- Sylvain has asked me to redo some work in the following ways:
   - Use React components for visualizations instead of using d3 to manipulate the DOM.
   - Allow creation of custom visualizations by allowing the user to supply variables (or formulae).

Feel free to set up a meeting to discuss, probably should involve Sylvain as well.

**Review by tashrifbillah — COMMENTED** [review]
It can be merged only if the work is compatible with https://github.com/PREDICT-DPACC/ethnicity-viz and PR #3

**Comment by blms** [comment]
> It can be merged only if the work is compatible with https://github.com/PREDICT-DPACC/ethnicity-viz and PR #3

It doesn't break compatibility with those. So ready to merge?

**Review by tashrifbillah — APPROVED** [review]

**Pull request merged**